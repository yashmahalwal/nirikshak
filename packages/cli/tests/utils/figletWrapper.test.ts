import figlet from "../../src/utils/figletWrapper";
describe("Figlet wraper", () => {
    test(`Valid input`, async () => {
        const result = await figlet("Lorem Ipsum");
        expect(result).toMatchSnapshot();
    });

    test(`Invalid input`, async () => {
        expect.hasAssertions();
        try {
            const out = await figlet("Lorem ipsum", {
                font: "Purpose" as any,
            });
            out;
        } catch (e) {
            expect(e).toMatchSnapshot();
        }
    });
});
