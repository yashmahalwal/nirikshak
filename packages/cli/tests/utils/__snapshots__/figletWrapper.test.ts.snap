// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Figlet wraper Invalid input 1`] = `[Error: ENOENT: no such file or directory, open '/home/yash/Desktop/nirikshak/packages/cli/node_modules/figlet/fonts/Purpose.flf']`;

exports[`Figlet wraper Valid input 1`] = `
"  _                               ___                           
 | |    ___  _ __ ___ _ __ ___   |_ _|_ __  ___ _   _ _ __ ___  
 | |   / _ \\\\| '__/ _ \\\\ '_ \` _ \\\\   | || '_ \\\\/ __| | | | '_ \` _ \\\\ 
 | |__| (_) | | |  __/ | | | | |  | || |_) \\\\__ \\\\ |_| | | | | | |
 |_____\\\\___/|_|  \\\\___|_| |_| |_| |___| .__/|___/\\\\__,_|_| |_| |_|
                                     |_|                        "
`;
